### DISCUSSION TITLE
clicking audio with ethernet

### DISCUSSION THREAD

**Discussion by parabyte** [pschatzmann/arduino-snapclient#29] [Q&A] [answered]
this is a issue i never had with wifi, i am having like a slight ticking sound that is regular. my brother suggested it could be a buffer issue he heard the issue too. I am really unsure how to diagnose or provide anything about this fault as there is  no error reports or problems to be reported from logs

**Comment by parabyte** [comment]
iv tried # define CONFIG_SNAPCAST_BUFF_LEN 2048 and it does not seem to help the issue, the issue seems to be some sort of memmory leak, if i power cycle my device the clicking does not accur initially, takes approx 8 minutes or so to happen, again this issue was not present when i was using wifi, so the issue is not on my desktop

**Comment by pschatzmann** [comment, marked as answer]
My guess would be that the Ethernet thruput is not fast enough for your selected settings.
- You can try to increase the [SPI frequency](https://github.com/espressif/arduino-esp32/blob/master/libraries/Ethernet/src/ETH.h) or
- optimize the settings in /etc/snapserver.conf

**Comment by parabyte** [comment]
i cant set any spi speed higher then 20mhz, im kinda thinking my mess of wires is not helping!

i am using  ... ETH.begin(ETH_PHY_TYPE, ETH_PHY_ADDR, ETH_PHY_CS, ETH_PHY_IRQ, ETH_PHY_RST, SPI,ETH_PHY_SPI_FREQ_MHZ);...

obviously putting 20.30.40 etc into the ETH_PHY_SPI_FREQ_MHZ variable.

so yes my slight audio problem is a bandwidth issue,  unfortunately i do not have a scope, can you suggest anything i can do to test out higher spi speeds? im not setting the vspi speed or passing anything to the spi  libraries directly just setting the spi speed via the ETH lib

**Comment by parabyte** [comment]
from my limited understanding, the max spi speed on the esp32 with a bi directional device like my ethernet module is limited to 26mhz and that should be enough bandwidth. i have played alot with the snap servers conf, if you can recommend any settings or anything i may of missed be really appreciated, i still have that ticking sound on audio playback and i refuse to revert back to wifi!

**Comment by parabyte** [comment]
i know for certain if i use flac obviously i wont have this issue cause of how flac works, but i do not wish to take the easy route, i really want pcm audio

**Comment by pschatzmann** [comment]
The most obvious would be to reduce the sample rate: You need to google the supported values for your selected codec!

**Comment by parabyte** [comment]
but i am  using pcm audio raw, I cannot really reduce the values for cd quality audio?

**Comment by parabyte** [comment]
as i said before, my setup worked perfectly on wifi, 
And the clicking is not imediate it happens after approx 8 or so minutes iv not timed it, a few tracks of audio.
i kinda think there might be some sort of bug in the espressif ethernet driver where its not freeing up after each packet? in its void ETHClass::end function? as i said before i am a complete novice with these things, but desperately want this ethernet to work perfectly.

**Reply by pschatzmann** [reply]
I bet you will not be able to hear the difference!

**Comment by parabyte** [comment]
okay this problem turned out to be a fault on my esp32 board,
